dev-utils/poly-cli/index.js

```diff
@@ -25,95 +25,43 @@ import { exit } from "process";
 
-if (existsSync("../dev-utils/rollup-plugin-copy-watch")) {
+if (fs.existsSync("../dev-utils/rollup-plugin-copy-watch")) {
     printInfoMsg("✓ Running from the right path!");
 } else {
     printErrorMsg(`This directory ${__dirname} will make the script fail
 Please change to «features»`);
     exit(1);
 }
 
-function setup(feature_name, author, version, description, license) {
-    // folders are objects, files are strings.
-    var structure = {};
+function setup(structure) {
+    try {
+        let feature_name = Object.keys(structure)[0];
 
-    // Remember "leaves" before subdirectories, or mkdir will fail
-    structure[feature_name] = {
-        src: {
-            "index.jsx": () =>
-                readFileSync(
-                    path.resolve(__dirname, "./src/static/templates/index.jsx")
-                ),
-            "styles.css": () =>
-                readFileSync(
-                    path.resolve(__dirname, "./src/static/templates/styles.css")
-                ),
-            locales: {
-                en: {
-                    "common.json": () =>
-                        readFileSync(
-                            path.resolve(
-                                __dirname,
-                                "./src/static/templates/locales/en/common.json"
-                            )
-                        ),
-                },
-                de: {
-                    "common.json": () =>
-                        readFileSync(
-                            path.resolve(
-                                __dirname,
-                                "./src/static/templates/locales/de/common.json"
-                            )
-                        ),
-                },
-            },
-            static: {
-                "manifest.json": () => manifestTemplate(feature_name, author),
-                "index.html": () =>
-                    readFileSync(
-                        path.resolve(
-                            __dirname,
-                            "./src/static/templates/index.html"
-                        )
-                    ),
-            },
-        },
-        test: [],
-        "package.json": () =>
-            packageTemplate(
-                feature_name,
-                version,
-                description,
-                "src/index.jsx",
-                author,
-                license
-            ),
-        "README.md": () => readmeTemplate(feature_name, description),
-        "rollup.config.mjs": () =>
-            readFileSync(
-                path.resolve(
-                    __dirname,
-                    "./src/static/templates/rollup.config.mjs"
-                )
-            ),
-    };
-
-    if (existsSync(`./${feature_name}`)) {
-        printErrorMsg("Feature already exists in this folder. Aborting!");
-        return;
-    }
+        if (fs.existsSync(`./${feature_name}`)) {
+            printErrorMsg("Feature already exists in this folder. Aborting!");
```

Comment:
Nit: You could throw an error here, then it will be handled at line 49.

Reply:
It will essentially exit if setup is not run. But yes, throwing an error will prevent complexity problems later on.